### PR TITLE
refactored unit model and added new units

### LINKED ISSUE #387: add new units to substance data in the editor
add "g/l" and "mg/l" as solution units
add "mol" and "mmol"

### PR DESCRIPTION
**Pull request by crapStone**
closes #387

### CODE REVIEW

File: src-tauri/cabr2_types/src/lib.rs [hunk 1]
```diff
@@ -66,17 +66,69 @@ pub struct Amount {
 #[serde(rename_all = "UPPERCASE")]
 pub enum Unit {
   Litre,
-  Milliliter,
+  Millilitre,
```
Comment (6543) [inline src-tauri/cabr2_types/src/lib.rs:69]: ...tre ?

Comment (Epsilon02) [reply src-tauri/cabr2_types/src/lib.rs:69]: Milliliter is american english.
Millilitre is british english.
So both will work I think.

Comment (Epsilon02) [reply src-tauri/cabr2_types/src/lib.rs:69]: Milliliter is american english.
Millilitre is british english.
So both will work I think.

Comment (crapStone) [reply src-tauri/cabr2_types/src/lib.rs]: both work, but i will change it back to american english

File: src/app/@core/models/substances.model.ts [hunk 2]
```diff
@@ -97,71 +106,137 @@ export interface Source {
...
+  PARTS_PER_MILLION,
+
+  CELSIUS,
+  FAHRENHEIT,
 }
 
-export interface GroupMapping<T> {
-  viewValue: string;
-  unitMappings: UnitMapping<T>[];
+// TODO move to i18n service
```
[the remark this answers is not in the document]
Comment (crapStone) [reply src/app/@core/models/substances.model.ts:152]: #396

**Review by Epsilon02 — DISMISSED** [review]

**Comment by Epsilon02** [comment]
Add ticket for model cleanup (remove group definitions)

**Review by Epsilon02 — APPROVED** [review]

**Pull request merged**